app/models/katello/content_view_puppet_module.rb

```diff
@@ -8,6 +8,7 @@ class ContentViewPuppetModule < Katello::Model
     validates :content_view_id, :presence => true
     validates :name, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
     validates :uuid, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
+    validate :uuid_associated_with_puppet_module
```

Comment:
[This before_filter](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb#L6) allows any of the puppet_module_params to be used to find the puppet module, given [my change](https://github.com/Katello/katello/pull/6093/files#diff-b3f64de16869054a88473910f7fef5cfR81) to that callback.